### PR TITLE
Add new <amp-script> tutorial

### PR DESCRIPTION
**Pull request by morsssss**
Adding code for the new <amp-script> tutorial. Please give feedback!

Note: this is based on https://github.com/CrystalOnScript/vanilla-js-amp-script.

### CODE REVIEW

File: amp-script-tutorial/finished_code/index.html [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+      .card {
+        background: #dee6ef;
+        padding: 1rem;
+        margin: .5rem;
+        border-radius: 8px;
+      }
+
+      .form-input {
+        margin-bottom: .5rem;
+      }
```
Comment (sebastianbenz) [inline amp-script-tutorial/finished_code/index.html]: suggestion: add

```
.form-input > *:first-child {
    width: 100px;
}
```

to align the form inputs

File: amp-script-tutorial/finished_code/js/script.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * If all the rules pass, enable the Submit button!
+ */
+  const checkPassword = () => {
+    const password = element.value;
+    let failed = false;
+
+    for (const check in checkRegexes) {
+      let li = document.getElementById(check);
+
+      if (password.match(checkRegexes[check])) {
```
Comment (sebastianbenz) [inline amp-script-tutorial/finished_code/js/script.js]: A simpler way to do this would be:

```
const updateStatus = (el, hasPassed) => {
  el.classList.toggle("invalid", !hasPassed);
  el.classList.toggle("valid", hasPassed);
}

...


const hasPassed = password.match(checkRegexes[check])
updateStatus(li, hasPassed)
```

It's less explicit though, so I'm not sure if that's favorable over the current solution.

File: amp-script-tutorial/starter_code/index.html [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        <div class="form-input">
+          <label>Email:</label>
+          <input type="text" name="email" tabindex=1 role="input" required>
+        </div>
+        <div class="form-input">
+          <label>Password:</label>
+          <input type=password 
+                name="password"
+                required
+                pattern="^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[^a-z\d]).{8,}$"
```
[the remark this answers is not in the document]
Comment (morsssss) [reply amp-script-tutorial/starter_code/index.html]: The pattern? Well, the original codelab started by showing how to make an input-checking experience without `<amp-script>`.  I kept that structure (though I did simplify the original regex slightly)

File: amp-script-tutorial/README.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+* `starter_code`: the code you start with
+* `finished_code`: what you'll end up with after you complete the tutorial
+
+Note that this code doesn't include a server, so if you actually submit the form, you'll get an error in the Console. This exercise is left for the reader 😎
+
+## Setup
+
+If your computer is already running a local server, you don't need to set anything up at all! You can just type the relevant URL into your browser. Depending on where you've put this code and how you've set up your server, that URL will look something like this:
+
+`http://localhost/amp-script-tutorial/starter_code/index.html`
```
[the remark this answers is not in the document]
Comment (morsssss) [reply amp-script-tutorial/README.md]: I've never needed that if I just spin up Apache on, say, a Mac.

File: amp-script-tutorial/finished_code/index.html (hunk 1 above)
Comment (morsssss) [reply amp-script-tutorial/finished_code/index.html]: Good idea!

How about this?

```css
      label {
        display: inline-block;
        width: 5rem;
      }
```

To me, it's simpler.

File: amp-script-tutorial/finished_code/js/script.js (hunk 2 above)
Comment (morsssss) [reply amp-script-tutorial/finished_code/js/script.js]: I struggled with this as I wrote it. I often struggle with this exact thing.

I think I prefer your version, though.

File: amp-script-tutorial/finished_code/index.html (hunk 1 above)
Comment (sebastianbenz) [reply amp-script-tutorial/finished_code/index.html]: absolutely

File: amp-script-tutorial/README.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# amp-script-tutorial
```
Comment (CrystalOnScript) [inline amp-script-tutorial/README.md]: I'm worried about using component naming dashes like this.  Might cause confusion.

File: amp-script-tutorial/README.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+## About
+
+This repository contains the code you'll use to do the [&lt;amp-script&gt; tutorial](https://amp.dev/documentation/guides-and-tutorials/develop/custom-javascript-tutorial/)!
```
Comment (CrystalOnScript) [inline amp-script-tutorial/README.md]: I'd say the name of the tutorial, not just "the amp-script" tutorial. We may have more?

File: amp-script-tutorial/README.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+This tutorial presents a form containing a password input. We want your password to follow certain rules, and so the user can't submit the form until those rules have been followed. In the tutorial, you begin with HTML that enforces the rules using &lt;amp-form&gt;'s `pattern` attribute. You then transform that into a more user-friendly experience which uses &lt;amp-script&gt;.
+
```
Comment (CrystalOnScript) [inline amp-script-tutorial/README.md]: Suggestion to remove passive voice:
```
This tutorial presents a form containing a password input. The password must meet certain criteria, enforced by <amp-form>'s pattern attribute. The user won't be unable to submit the form without meeting this criteria. You will implement functionality with <amp-script> to improve the user's experience. 
```

File: amp-script-tutorial/README.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+This repo contains two subdirectories:
+* `starter_code`: the code you start with
+* `finished_code`: what you'll end up with after you complete the tutorial
+
+Note that this code doesn't include a server, so if you actually submit the form, you'll get an error in the Console. This exercise is left for the reader 😎
```
Comment (CrystalOnScript) [inline amp-script-tutorial/README.md]: Suggestion to change `This exercise is left for the reader 😎`  to something like `Use your preferred method or one of the setup suggestions below to spin up a sever.`

File: amp-script-tutorial/README.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+## Setup
+
+If your computer is already running a local server, you don't need to set anything up at all! You can just type the relevant URL into your browser. Depending on where you've put this code and how you've set up your server, that URL will look something like this:
```
Comment (CrystalOnScript) [inline amp-script-tutorial/README.md]: Avoid words such as "just"

File: amp-script-tutorial/README.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+`http://localhost/amp-script-tutorial/starter_code/index.html`
+
+Alternately, you can set up a simple local server using something like [serve](https://www.npmjs.com/package/serve), the node.js-based static content server.
+
+Assuming you've installed [node](https://nodejs.org/), you can simply go to your command line, enter the directory where your code lives, and type:
```
Comment (CrystalOnScript) [inline amp-script-tutorial/README.md]: Cut use of "simple" or "simply".

File: amp-script-tutorial/README.md [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+Note that this code doesn't include a server, so if you actually submit the form, you'll get an error in the Console. This exercise is left for the reader 😎
+
+## Setup
+
+If your computer is already running a local server, you don't need to set anything up at all! You can just type the relevant URL into your browser. Depending on where you've put this code and how you've set up your server, that URL will look something like this:
+
+`http://localhost/amp-script-tutorial/starter_code/index.html`
+
+Alternately, you can set up a simple local server using something like [serve](https://www.npmjs.com/package/serve), the node.js-based static content server.
```
Comment (CrystalOnScript) [inline amp-script-tutorial/README.md]: I think Node.js is a proper noun and should be capitalized.

File: amp-script-tutorial/finished_code/js/script.js (hunk 2 above)
Comment (morsssss) [reply amp-script-tutorial/finished_code/js/script.js]: Actually, I'm seeing that this signature for `toggle` isn't supported in IE. (Which may be why I'd never heard of it.) So I think it would require a more complex construction for me to take this more compact approach.

Which is too bad, because I also wanted to update the `failed` flag using a Boolean operator. How often do you get to use one of those?

File: amp-script-tutorial/README.md (hunk 5 above)
Comment (morsssss) [reply amp-script-tutorial/README.md]: It is a little confusing, isn't it? But I'm just following the naming convention used in this repo, and in other projects' README's.

File: amp-script-tutorial/README.md (hunk 6 above)
Comment (morsssss) [reply amp-script-tutorial/README.md]: I doubt that we will have more. But... yes, I see your point.

I was calling this "Use JavaScript in AMP". How would you phrase this, then? I'd normally say something like, "the &lt;amp-script&gt; tutorial: 'Use JavaScript in AMP.'" Would you say, instead, "the 'Use JavaScript in AMP' tutorial'?

File: amp-script-tutorial/README.md (hunk 8 above)
Comment (morsssss) [reply amp-script-tutorial/README.md]: I'd have to also instruct the reader to not just create a server, as in the original tutorial, but also to include a service that handles submissions (which the original tutorial doesn't do). I'd rather keep this short and sweet unless you think I should provide this explicit guidance.

File: amp-script-tutorial/README.md (hunk 9 above)
Comment (morsssss) [reply amp-script-tutorial/README.md]: Ok... what do you suggest?

"You can simply type..."?

"All you need to do is type..."?

Something even more brief?

File: amp-script-tutorial/README.md (hunk 10 above)
Comment (morsssss) [reply amp-script-tutorial/README.md]: oo, thanks. Done.

File: amp-script-tutorial/README.md (hunk 11 above)
Comment (morsssss) [reply amp-script-tutorial/README.md]: I always put it in lowercase, and that's how their logo looks... but, you're right. Thanks. Capitalized!

File: amp-script-tutorial/README.md (hunk 7 above)
Comment (morsssss) [reply amp-script-tutorial/README.md]: Right. I should know that :)

File: amp-script-tutorial/README.md (hunk 6 above)
Comment (CrystalOnScript) [reply amp-script-tutorial/README.md]: Maybe "This repository contains the code you need to follow the Build custom UI with amp-script tutorial"

File: amp-script-tutorial/README.md (hunk 8 above)
Comment (CrystalOnScript) [reply amp-script-tutorial/README.md]: A tutorial should provide every single step the follower needs to succeed - as in I should never have to follow a different tutorial to successfully reach the end project. So I'd say provide explicit guidance in this case.

File: amp-script-tutorial/README.md (hunk 9 above)
Comment (CrystalOnScript) [reply amp-script-tutorial/README.md]: Remove the word "just" is probably fine. I would reword it as:
`Instead, you can type the relevant URL into your browser.`

File: amp-script-tutorial/README.md [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+
+## Setup
+
+If your computer is already running a local server, you don't need to set anything up at all! You can just type the relevant URL into your browser. Depending on where you've put this code and how you've set up your server, that URL will look something like this:
+
+`http://localhost/amp-script-tutorial/starter_code/index.html`
+
+Alternately, you can set up a quick local server using something like [serve](https://www.npmjs.com/package/serve), the Node.js-based static content server.
+
+Assuming you've installed [node](https://nodejs.org/), you can simply go to your command line, enter the directory where your code lives, and type:
```
Comment (CrystalOnScript) [inline amp-script-tutorial/README.md]: Please remove `simply` :)

File: amp-script-tutorial/README.md (hunk 5 above)
Comment (CrystalOnScript) [reply amp-script-tutorial/README.md]: Ahh, my bad. Nah, totes fine. Ignore :)

File: amp-script-tutorial/README.md (hunk 12 above)
Comment (morsssss) [reply amp-script-tutorial/README.md]: Ah, I thought you meant one or the other. You meant both!

There's probably a basic rule of writing here I'm not aware of. Perhaps, simply, remove unnecessary words? Or... should I say... "remove extra words"?

Comment (CrystalOnScript) [reply amp-script-tutorial/README.md]: Using words like "simply" and "just" assume these things will be easy for the reader when they may not be. Google's style guide does specifically point out using "simply": https://developers.google.com/style/tone?hl=pt-br

File: amp-script-tutorial/README.md (hunk 6 above)
Comment (morsssss) [reply amp-script-tutorial/README.md]: ok! (though I have also changed the title)

File: amp-script-tutorial/README.md (hunk 8 above)
Comment (morsssss) [reply amp-script-tutorial/README.md]: ok!

File: amp-script-tutorial/README.md (hunk 9 above)
Comment (morsssss) [reply amp-script-tutorial/README.md]: ok!

File: amp-script-tutorial/README.md (hunk 12 above)
Comment (morsssss) [reply amp-script-tutorial/README.md]: I'm reading that... this is very interesting!

The word "just" is like a reflex for me. I've removed it, but as I re-edit it's so hard to avoid just sticking it back in. (see, there it is again!)

Comment (CrystalOnScript) [reply amp-script-tutorial/README.md]: Yeah, its sneaky! And it totally sounds like a good thing to be saying - weird huh?

**Comment by morsssss** [comment]
Let's push this out at the same time as the new tutorial. I'll link that PR here when it's ready.
